flink-table/flink-table-planner-blink/src/test/java/org/apache/flink/table/planner/factories/TestValuesTableFactory.java

```diff
@@ -291,6 +291,14 @@ private static RowKind parseRowKind(String rowKindShortString) {
 			"Optional map of 'metadata_key:data_type'. The order will be alphabetically. " +
 			"The metadata is part of the data when enabled.");
 
+	private static final ConfigOption<Integer> SINK_INDEX_OF_ROWTIME = ConfigOptions
+		.key("sink-index-of-rowtime")
+		.intType()
+		.defaultValue(-1)
+		.withDeprecatedKeys(
+			"Option index of the rowtime field. The default value -1 indicate that don't drop " +
+			"the late data.");
```

Comment:
If this option controls whether to drop late data, then the option name should be something like `sink.drop-late-event=true`.

The rowtime field can be derived from WATERMARK statement (can also be declared on sink), instead of a redundant option. 

Besides, I find you only support this option on append only sink, then you should throw exceptions if the sink is not append only during `createDynamicTableSink`.